### PR TITLE
CLOUDP-60953: Implement download diagnose archive API, OM client

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ [CLOUDP-60953](https://jira.mongodb.org/browse/CLOUDP-60953)

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] I have tested my code using my local OM

### CODE REVIEW

File: opsmngr/diagnostics.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+)
+
+// DiagnosticsService is an interface for interfacing with Diagnostic Archives in MongoDB Ops Manager APIs
+// https://docs.opsmanager.mongodb.com/current/reference/api/diagnostic-archives/
+type DiagnosticsService interface {
+	Get(context.Context, string, io.Writer) (*atlas.Response, error)
+}
+
+type DiagnosticsServiceOp struct {
+	Client atlas.GZipRequestDoer
```
Comment (andreaangiolillo) [inline opsmngr/diagnostics.go:37]: We have to change GZipRequestDoer to accept a body if we want to allow the client to set the query params `limit` and `minutes`.

Comment (andreaangiolillo) [reply opsmngr/diagnostics.go:37]: Do we want to do it as part of this PR? @gssbzn

File: opsmngr/diagnostics.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+// Get retrieves the project’s diagnostics archive file.
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/diagnostics/get-project-diagnostic-archive/
+func (s *DiagnosticsServiceOp) Get(ctx context.Context, projectID string, out io.Writer) (*atlas.Response, error) {
+	path := fmt.Sprintf(diagnosticsBasePath, projectID)
```
Comment (gssbzn) [inline opsmngr/diagnostics.go:43]: can you please add the validation for project id here?

File: opsmngr/diagnostics.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+type DiagnosticsService interface {
+	Get(context.Context, string, *DiagnosticsListOpts, io.Writer) (*atlas.Response, error)
+}
+
+type DiagnosticsServiceOp struct {
+	Client atlas.GZipRequestDoer
+}
+
+type DiagnosticsListOpts struct {
+	Limit   int64 `json:"limit"`
```
Comment (gssbzn) [inline opsmngr/diagnostics.go:41]: good call on adding this but let's "ommitempty"

Comment (gssbzn) [reply opsmngr/diagnostics.go:41]: same for minutes

**Pull request merged**